src/main/java/ca/fxco/pistonlib/blocks/pistons/configurablePiston/ConfigurableMovingBlock.java

```diff
@@ -170,14 +140,11 @@ public void tick(BlockState state, ServerLevel level, BlockPos pos, RandomSource
 
     @Override
     public boolean canSurvive(BlockState state, LevelReader level, BlockPos pos) {
-        return !slippery || BaseSlipperyBlock.calculateDistance(level, pos) < MAX_DISTANCE;
+        return !this.family.isSlippery() || BaseSlipperyBlock.calculateDistance(level, pos) < MAX_DISTANCE;
     }
 
     @Override
     public void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
-        builder.add(FACING, TYPE);
-        if (slippery) {
-            builder.add(SLIPPERY_DISTANCE);
-        }
+        builder.add(FACING, TYPE, SLIPPERY_DISTANCE);
```

Comment:
Why do all configurable pistons now have `SLIPPERY_DISTANCE`?

Reply:
Yeah that's a bit annoying. It's because the `family` field has not yet been assigned when `createBlockStateDefinition` is called.

Reply:
ohhh, ok thats going to be a pain when I'm going to add more stuff to the configurable piston. I'll have to look into that